pages/components/cards.js

```diff
@@ -82,33 +81,55 @@ export default function Cards({ posts, setPosts }) {
     }, 1000);
   };
 
-  // Delete post
-  const handleDeletePost = (postId) => {
-    const updatedPosts = posts.filter((post) => post.id !== postId);
+  // Edit comment
+  const handleEditComment = (postId) => {
+    const updatedPosts = posts.map((post) => {
+      if (post.id === postId) {
+        return {
+          ...post,
+          isEditing: true,
+          editedContent: post.content,
+        };
+      }
+      return post;
+    });
     setPosts(updatedPosts);
   };
 
-  // Edit post content
-  const handleContentEdit = (postId, content) => {
-    setEditContent(content);
-    setEditPostId(postId);
-  };
-
-  // Update post content
-  const handleContentUpdate = (event, postId) => {
+  // Save edited comment
+  const handleSaveComment = (event, postId) => {
     event.preventDefault();
     const updatedPosts = posts.map((post) => {
       if (post.id === postId) {
         return {
           ...post,
-          content: editContent,
+          isEditing: false,
+          content: post.editedContent,
         };
       }
       return post;
     });
     setPosts(updatedPosts);
-    setEditContent("");
-    setEditPostId(null);
+  };
+
+  // Delete post
+  const handleDeletePost = (postId) => {
+    const updatedPosts = posts.filter((post) => post.id !== postId);
+    setPosts(updatedPosts);
+  };
+
+  const calculateLikeCount = (post) => {
+    if (post && post.likes) {
+      return post.likes;
+    }
+    return 0;
+  };
+
+  const calculateCommentCount = (post) => {
+    if (post && post.comments) {
+      return post.comments.length;
+    }
+    return 0;
   };
```

Comment:
this looks to be more work than necessary.

Why not when creating a new post, the post object has a key `likes` that starts at 0, and a key `comments` that is an empty array.
You would not need those functions anymore but could instead derive the values directly from your state

Reply:
there will still be a database connection. It is only solved for now so that I can see how it works.